src/parser.rs

```diff
@@ -15,90 +65,97 @@ fn is_digit(c: char) -> bool {
     }
 }
 
-named!(letter<CompleteStr, char>,
+named!(letter<Span, char>,
     one_of!("ABCDEFGHIJKLMNOPQRSTUVWXYZ"));
 
-named!(digit<CompleteStr, u8>,
+named!(digit<Span, u8>,
     map!(one_of!("0123456789"), |c| c as u8 - b'0'));
 
 // 5. Programs
 
-named!(pub program<CompleteStr, Result<Program, Error>>,
+named!(pub program<Span, Result<Program, Error>>,
     do_parse!(
         blocks: many0!(terminated!(block, end_of_line)) >>
         opt!(end_of_line) >>
         (Program::new(blocks))
     ));
 
-named!(pub block<CompleteStr, Block>,
+named!(block<Span, Block>,
     do_parse!(
         line_number: line_number >>
         statement: sep!(space, statement) >>
         space0 >>
-        (Block::Line{ line_number, statement })
+        (Block::Line{
+            line_number,
+            statement: statement.0,
+            statement_source: statement.1
+        })
     ));
 
-named!(line_number<CompleteStr, u16>,
+named!(line_number<Span, u16>,
     map_res!(take_while_m_n!(1, 4, is_digit), from_decimal));
 
-named!(end_of_line<CompleteStr, CompleteStr>, alt!(eof!() | eol));
+named!(end_of_line<Span, Span>, alt!(eof!() | eol));
 
-named!(end_statement<CompleteStr, Statement>,
+named!(end_statement<Span, Statement>,
     do_parse!(
         tag!("END") >>
         (Statement::End)
     ));
 
-named!(statement<CompleteStr, Statement>,
-    alt!(
-        goto_statement |
-        gosub_statement |
-        if_then_statement |
-        let_statement |
-        print_statement |
-        return_statement |
-        stop_statement |
-        remark_statement |
-        end_statement
+named!(statement<Span, (Statement, Span)>,
+    do_parse!(
+        statement_source: position!() >>
+        statement: alt!(
+            goto_statement |
+            gosub_statement |
+            if_then_statement |
+            let_statement |
+            print_statement |
+            return_statement |
+            stop_statement |
+            remark_statement |
+            end_statement) >>
+        (statement, statement_source)
```

Comment:
See above. What type is `statement_source`. Is it the position or the original source string?

Reply:
Its `Span`. Cf. also `position!()` call in https://docs.rs/nom_locate/0.3.1/nom_locate/#how-to-use-it